### ISSUE TITLE
Why is the sigmoid function applied to logits before calling F.binary_cross_entropy_with_logits?

### ISSUE THREAD

**Issue by kirill-menke** [facebookresearch/fvcore#126]
In [focal_loss.py](https://github.com/facebookresearch/fvcore/blob/main/fvcore/nn/focal_loss.py) you can find the following code snipped in `sigmoid_focal_loss` function:

https://github.com/facebookresearch/fvcore/blob/fd5043ff8b2e6790f5bd7c9632695c68986cc658/fvcore/nn/focal_loss.py#L35-L36

Why is the sigmoid function applied to the inputs before calculating the loss? Doesn't `F.binary_cross_entropy_with_logits` expect the raw logits?
